apps/condo/domains/resident/schema/RegisterResidentService.js

```diff
@@ -35,8 +41,20 @@ const RegisterResidentService = new GQLCustomSchema('RegisterResidentService', {
                     unitType,
                     user: { connect: { id: context.authedItem.id } },
                 }
+
+                // TODO(AleX83Xpert): Think about how to prevent copying this condition :-/
+                const client = conf.NODE_ENV === 'test' || get(conf, 'ADDRESS_SERVICE_CLIENT_MODE') === 'fake'
```

Comment:
You can move this logic to createAddressServiceClientInstance function with all checks for NODE_ENV and ADDRESS_SERVICE_CLIENT_MODE
Reasons:
1. RegisterResidentService will have less code and checks (for address service environment)
2. Now you  need to copy-paste this logic to every place where you is using addressService

Reply:
Unfortunately, functions have different signatures :(
The real client needs only the address service URL. The test client needs an object.

Reply:
But you can change it